Makefile

```diff
@@ -1,66 +1,53 @@
 MAKEFILE_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 
-PACKER_DEFAULT_VARIABLE_FILE ?= $(MAKEFILE_DIR)/eks-worker-al2-variables.json
-PACKER_TEMPLATE_FILE ?= $(MAKEFILE_DIR)/eks-worker-al2.json
-PACKER_BINARY ?= packer
-AVAILABLE_PACKER_VARIABLES := $(shell $(PACKER_BINARY) inspect -machine-readable $(PACKER_TEMPLATE_FILE) | grep 'template-variable' | awk -F ',' '{print $$4}')
+# `kubernetes_version` is a build variable, but requires some introspection
+# to dynamically pull determine build templates & variable defaults.
+# initialize the kubernetes version from the provided variables if missing.
+ifeq ($(kubernetes_version),)
+ifneq ($(PACKER_VARIABLE_FILE),)
+	kubernetes_version ?= $(shell jq -r .kubernetes_version $(PACKER_VARIABLE_FILE))
+endif
+endif
 
 K8S_VERSION_PARTS := $(subst ., ,$(kubernetes_version))
 K8S_VERSION_MINOR := $(word 1,${K8S_VERSION_PARTS}).$(word 2,${K8S_VERSION_PARTS})
 
-# expands to 'true' if PACKER_VARIABLE_FILE is non-empty
-# and the file contains the string passed as the first argument
-# otherwise, expands to 'false'
-packer_variable_file_contains = $(if $(PACKER_VARIABLE_FILE),$(shell grep -Fq $1 $(PACKER_VARIABLE_FILE) && echo true || echo false),false)
-
-# expands to 'true' if the version comparison is affirmative
-# otherwise expands to 'false'
-vercmp = $(shell $(MAKEFILE_DIR)/files/bin/vercmp "$1" "$2" "$3")
-
-# Docker is not present on 1.25+ AMI's
-# TODO: remove this when 1.24 reaches EOL
-ifeq ($(call vercmp,$(kubernetes_version),gteq,1.25.0), true)
-	# do not tag the AMI with the Docker version
-	docker_version ?= none
-	# do not include the Docker version in the AMI description
-	ami_component_description ?= (k8s: {{ user `kubernetes_version` }}, containerd: {{ user `containerd_version` }})
+al_variant ?= al2
```

Comment:
I think this variable could be a bit more generic, such as `OS_DISTRO`. We could feasibly add `ubuntu22.04`, `rhel9`, etc.  or something that isn't Amazon Linux in the future.

Reply:
I mean, Amazon Linux *is* an OS 😉 We could use `al_version`, but then I'd expect values like `2` and `2023`. Maybe `base_distro`?

Reply:
`os_distro` has grown on me the more i fiddle with names 🤔, ill try rolling with it